### PR TITLE
Add section on managing metadata

### PR DESCRIPTION
**Pull request by J0**
There have been many queries about metadata management which suggests that it's probably worth adding a section about it. 

Some issues about it - 

[How to get Metadata](https://github.com/supabase/gotrue-js/issues/576)
[OAuth User metadata](https://github.com/supabase/gotrue-js/issues/457) -> This is still [not merged yet](https://github.com/supabase/gotrue-js/pull/611) but it falls under the same category of metadata management
[Discord metadata](https://github.com/supabase/gotrue-js/issues/453)
[Security definer metadata](https://github.com/supabase/gotrue/issues/975) -> Needs verification

### CODE REVIEW

File: apps/docs/pages/guides/auth/managing-user-data.mdx [hunk 1]
```diff
@@ -87,6 +87,36 @@ If you need to fetch a full list of user profiles, we supply a `service_key` whi
 
 Make sure you _NEVER_ expose this publicly. But it can be used on the server-side to fetch all of the profiles.
 
+## Accessing User Metadata
+
+Users can pass in metadata on sign up. User metadata is stored on the `raw_user_meta_data` column of the `auth.users` table.
+
+Here's an example of how to add metadata fields on signup:
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/managing-user-data.mdx:94]:
```suggestion
You can assign metadata to users on sign up:
```

File: apps/docs/pages/guides/auth/managing-user-data.mdx [hunk 2]
```diff
@@ -87,6 +87,36 @@ If you need to fetch a full list of user profiles, we supply a `service_key` whi
...
+      data: {
+        first_name: 'John',
+        age: 27,
+      }
+    }
+  }
+)
+```
+
+To view the metadata:
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/managing-user-data.mdx:111]:
```suggestion
User metadata is stored on the `raw_user_meta_data` column of the `auth.users` table. To view the metadata:
```

File: apps/docs/pages/guides/auth/managing-user-data.mdx [hunk 3]
```diff
@@ -87,6 +87,36 @@ If you need to fetch a full list of user profiles, we supply a `service_key` whi
 
 Make sure you _NEVER_ expose this publicly. But it can be used on the server-side to fetch all of the profiles.
 
+## Accessing User Metadata
+
+Users can pass in metadata on sign up. User metadata is stored on the `raw_user_meta_data` column of the `auth.users` table.
+
+Here's an example of how to add metadata fields on signup:
+
+```
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/managing-user-data.mdx:96]:
```suggestion
```js
```

File: apps/docs/pages/guides/auth/managing-user-data.mdx [hunk 4]
```diff
@@ -87,6 +87,36 @@ If you need to fetch a full list of user profiles, we supply a `service_key` whi
 
 Make sure you _NEVER_ expose this publicly. But it can be used on the server-side to fetch all of the profiles.
 
+## Accessing User Metadata
+
+Users can pass in metadata on sign up. User metadata is stored on the `raw_user_meta_data` column of the `auth.users` table.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/managing-user-data.mdx:92]: We can remove this one now, as it's duplicated in the suggestions below

File: apps/docs/pages/guides/auth/managing-user-data.mdx [hunk 5]
```diff
@@ -87,6 +87,36 @@ If you need to fetch a full list of user profiles, we supply a `service_key` whi
...
+```
+
+To view the metadata:
+
+```js
+const {
+  data: { user },
+} = await supabase.auth.getUser()
+let metadata = user.user_metadata
+```
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/managing-user-data.mdx:118]: Consider adding tabs here, one with:

select raw_user_meta_data
from auth.users;

**Review by kiwicopple — APPROVED** [review]

**Review by kangmingtay — APPROVED** [review]
Probably also a good idea to add docs on how to add user_metadata for the `admin.createUser` and `admin.updateUser` methods

File: apps/docs/pages/guides/auth/managing-user-data.mdx (hunk 5 above)
Comment (J0) [reply apps/docs/pages/guides/auth/managing-user-data.mdx:114]: Will add this in the FLUP with section about admin updates

**Comment by J0** [comment]
Related issues to address in FLUP: 

Triggers on insert might not capture metadata  because an insert call is made for user and then a follow up call is made to update user metadata- https://github.com/supabase/gotrue/issues/975

**Pull request merged**